modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/migrator/ClaimDataMigrator.java

```diff
@@ -0,0 +1,296 @@
+            existingExternalClaimURIs.get(claimDialectURI).add(claimURI);
+        }
+    }
+
+    /**
+     * Get existing local claim URIs.
+     *
+     * @param tenantId tenant id
+     * @return existing claim URI set
+     * @throws ClaimMetadataException ClaimMetadataException
+     */
+    private Set<String> getExistingLocalClaimURIs(int tenantId) throws ClaimMetadataException {
+
+        Set<String> localClaimURIs = new HashSet<>();
+        for (LocalClaim localClaim : localClaimDAO.getLocalClaims(tenantId)) {
+            localClaimURIs.add(localClaim.getClaimURI());
+        }
+        return localClaimURIs;
+    }
+
+    /**
+     * Get existing external claim URIs.
+     *
+     * @param tenantId        tenant id
+     * @param claimDialectURI claim dialect URI
+     * @return existing external claim URIs
+     * @throws ClaimMetadataException ClaimMetadataException
+     */
+    private Set<String> getExistingExternalClaimURIs(int tenantId, String claimDialectURI)
+            throws ClaimMetadataException {
+
+        Set<String> externalClaimURIs = new HashSet<>();
+        for (ExternalClaim externalClaim : externalClaimDAO.getExternalClaims(claimDialectURI, tenantId)) {
+            externalClaimURIs.add(externalClaim.getClaimURI());
+        }
+        return externalClaimURIs;
+    }
+
+    /**
+     * Add local claim mapping.
+     *
+     * @param tenantId          tenant id
+     * @param primaryDomainName primary domain name
+     * @param claimURI          claim URI
+     * @param claimMapping      claim mappings
+     * @throws ClaimMetadataException ClaimMetadataException
+     */
+    private void addLocalClaimMapping(int tenantId, String primaryDomainName, String claimURI, org.wso2.carbon.user
+            .core.claim.ClaimMapping claimMapping) throws ClaimMetadataException {
+
+        List<AttributeMapping> mappedAttributes = new ArrayList<>();
+        if (StringUtils.isNotBlank(claimMapping.getMappedAttribute())) {
+            mappedAttributes.add(new AttributeMapping(primaryDomainName, claimMapping.getMappedAttribute()));
+        }
+        if (claimMapping.getMappedAttributes() != null) {
+            for (Map.Entry<String, String> claimMappingEntry : claimMapping.getMappedAttributes().entrySet()) {
+                mappedAttributes.add(new AttributeMapping(claimMappingEntry.getKey(), claimMappingEntry.getValue()));
+            }
+        }
+
+        Map<String, String> claimProperties = claimConfig.getPropertyHolder().get(claimURI);
+        claimProperties.remove(ClaimConstants.DIALECT_PROPERTY);
+        claimProperties.remove(ClaimConstants.CLAIM_URI_PROPERTY);
+        claimProperties.remove(ClaimConstants.ATTRIBUTE_ID_PROPERTY);
+
+        if (!claimProperties.containsKey(ClaimConstants.DISPLAY_NAME_PROPERTY)) {
+            claimProperties.put(ClaimConstants.DISPLAY_NAME_PROPERTY, "0");
+        }
+        if (claimProperties.containsKey(ClaimConstants.SUPPORTED_BY_DEFAULT_PROPERTY) &&
```

Comment:
Is this the correct logic for setting supported_by_default, read_only and required to true? Check and verify

Reply:
I checked with the DefaultClaimMetadataStore implementation. This same logic is used